src/arrayMethodSort.js

```diff
@@ -4,8 +4,32 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      for (let i = 0; i < this.length - 1; i++) {
+        if (this[i] + '' > this[i + 1] + '') {
+          const currentIndex = this[i];
+
+          this[i] = this[i + 1];
+          this[i + 1] = currentIndex;
+
+          i = -1;
```

Comment:
Resetting the index to -1 after a swap is inefficient as it causes the algorithm to re-check all elements, even those that have already been sorted. Consider using a different sorting algorithm or optimizing the current one.